src/raftstore/store/store.rs

```diff
@@ -1691,17 +1691,14 @@ impl<T: Transport, C: PdClient> Store<T, C> {
     }
 
     fn handle_snap_mgr_gc(&mut self) -> Result<()> {
-        let mut snap_keys = try!(self.snap_mgr.list_snap());
+        let mut snap_keys = try!(self.snap_mgr.list_idle_snap());
```

Comment:
Thinking of move this loop to be a method of `SnapManager`, so that the code is protected by a single call to lock.